bitcoin/src/address/error.rs

```diff
@@ -71,9 +73,7 @@ impl fmt::Display for UnknownAddressTypeError {
 }
 
 #[cfg(feature = "std")]
-impl std::error::Error for UnknownAddressTypeError {
-    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> { None }
-}
+impl_sourceless_error!(UnknownAddressTypeError);
```

Comment:
FTR the current preferred style for these is 

```rust
impl std::error::Error for UnknownAddressTypeError {}
```

Which is exactly the same as the current code because default impl returns `None`. In days gone by I used the more explicit style but in hindsight I was wrong.

Reply:
Yeah, I don't think there's much benefit to using the explicit style. As long as there's a visible impl we'll (hopefully) notice it and update it when we add variants.